tests/integration/utils.js

```diff
@@ -0,0 +1,91 @@
+/**
+ * @fileoverview Utilities for running integration tests
+ * @author James Henry <https://github.com/JamesHenry>
+ * @copyright jQuery Foundation and other contributors, https://jquery.org/
+ * MIT License
+ */
+"use strict";
+
+//------------------------------------------------------------------------------
+// Requirements
+//------------------------------------------------------------------------------
+const eslint = require("eslint");
+const unpad = require("dedent");
+const path = require("path");
+
+//------------------------------------------------------------------------------
+// Public
+//------------------------------------------------------------------------------
+
+/* eslint-disable no-use-before-define */
+module.exports = {
+    verifyAndAssertMessages
+};
+/* eslint-enable no-use-before-define */
+
+/**
+ * A utility function to lint the given source code using typescript-eslint-parser
+ * and assert expectations around the linting feedback
+ *
+ * @param {string} code source code to lint
+ * @param {Object} rules ESLint rules configuration
+ * @param {string[]} expectedMessages Expected linting feedback
+ * @param {string} sourceType Either "module" or "script"
+ * @param {Obejct} overrideConfig Custom configuration options for the CLIEngine
+ * @returns {void}
+ */
+function verifyAndAssertMessages(
+    code,
+    rules,
+    expectedMessages,
+    sourceType,
+    overrideConfig
+) {
+    const cliEngineConfig = {
+        useEslintrc: false,
+        parser: require.resolve("../../parser"),
+        rules,
+        parserOptions: {
+            ecmaVersion: 8,
+            ecmaFeatures: {
+                jsx: true,
+                experimentalObjectRestSpread: true,
+                globalReturn: true
+            },
+            sourceType: sourceType || "script"
+        },
+        cwd: path.join(__dirname)
+    };
+
+    if (overrideConfig) {
+        Object.assign(cliEngineConfig, overrideConfig);
+    }
+
+    const cli = new eslint.CLIEngine(cliEngineConfig);
```

Comment:
I think it would be possible to just use `eslint.Linter` here:

```js
const linter = new eslint.Linter();

linter.defineParser("typescript-eslint-parser", require("../../parser"));
linter.defineRules(rules);
const messages = linter.verify(
  code,
  {
    parser: 'typescript-eslint-parser',
    parserOptions: {
      ecmaVersion: 8,
      ecmaFeatures: {
        jsx: true,
        experimentalObjectRestSpread: true,
        globalReturn: true
      },
      sourceType: sourceType || "script"
    }
  },
  { filename: 'fake-filename.js' }
)
```

Reply:
Unless I am mistaken when using the Linter, it is not possible to use `extends`. Being able to run integration tests concisely using presets is important IMO.

Reply:
You're correct that it's not possible to use `extends` with Linter, but it doesn't seem like `extends` is being used for any of these tests. Were you planning to add it?

(I don't mean to block the PR with this concern -- I'm just bringing it up in case you made a mistake.)

Reply:
The one I pasted in directly above, line 20?

Reply:
Oops, I missed that, sorry. (I hadn't realized that this function was also being used from other files.)

Reply:
Yeah I wasn't a huge fan of the huge single file used by babel-eslint so I extracted it and refactored it within a helper function.

Thanks for the review!